pkg/controllers/user/pipeline/controller/pipeline/pipeline.go

```diff
@@ -0,0 +1,155 @@
+package pipeline
+
+import (
+	"context"
+	"errors"
+	"github.com/rancher/rancher/pkg/controllers/user/pipeline/remote"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/satori/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+type Lifecycle struct {
+	pipelines                  v3.PipelineInterface
+	pipelineLister             v3.PipelineLister
+	sourceCodeCredentialLister v3.SourceCodeCredentialLister
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	pipelines := cluster.Management.Management.Pipelines("")
+	pipelineLister := pipelines.Controller().Lister()
+	pipelineExecutions := cluster.Management.Management.PipelineExecutions("")
+	pipelineExecutionLogs := cluster.Management.Management.PipelineExecutionLogs("")
+	sourceCodeCredentialLister := cluster.Management.Management.SourceCodeCredentials("").Controller().Lister()
+
+	pipelineLifecycle := &Lifecycle{
+		pipelines:                  pipelines,
+		pipelineLister:             pipelineLister,
+		sourceCodeCredentialLister: sourceCodeCredentialLister,
+	}
+	s := &CronSyncer{
+		pipelineLister:        pipelineLister,
+		pipelines:             pipelines,
+		pipelineExecutions:    pipelineExecutions,
+		pipelienExecutionLogs: pipelineExecutionLogs,
+	}
+
+	pipelines.AddLifecycle("pipeline-controller", pipelineLifecycle)
+	go s.sync(ctx, syncInterval)
+}
+
+func (l *Lifecycle) Create(obj *v3.Pipeline) (*v3.Pipeline, error) {
+
+	obj.Status.NextRun = 1
+	obj.Status.State = "active"
```

Comment:
this can be done by setting norman default tag in schema in rancher/types

Reply:
yep, but seems that the default value is not return on create request so set them here to make sure front end renders correctly